osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -583,6 +574,70 @@ protected override void Dispose(bool isDisposing)
             hitEvents.Clear();
         }
 
+        #region Static helper methods
+
+        /// <summary>
+        /// Given an accuracy (0..1), return the correct <see cref="ScoreRank"/>.
+        /// </summary>
+        public static ScoreRank RankFromAccuracy(double accuracy)
+        {
+            switch (accuracy)
+            {
+                case accuracy_cutoff_x:
+                    return ScoreRank.X;
+
+                case >= accuracy_cutoff_s:
+                    return ScoreRank.S;
+
+                case >= accuracy_cutoff_a:
+                    return ScoreRank.A;
+
+                case >= accuracy_cutoff_b:
+                    return ScoreRank.B;
+
+                case >= accuracy_cutoff_c:
+                    return ScoreRank.C;
+
+                default:
+                    return ScoreRank.D;
+            }
+        }
+
+        /// <summary>
+        /// Given a <see cref="ScoreRank"/>, return the cutoff accuracy (0..1).
+        /// Accuracy must be great or equal to the cutoff to qualify for the provided rank.
```

Comment:
```suggestion
        /// Given a <see cref="ScoreRank"/>, return the cutoff accuracy (0..1).
        /// Accuracy must be greater or equal to the cutoff to qualify for the provided rank.
```

Reply:
not sure why you didn't just push that :thinking:

Reply:
i'm glad i didn't because i didn't even make it correct